src/panels/config/mqtt/mqtt-device-debug-info-card.ts

```diff
@@ -0,0 +1,174 @@
+  try {
+    const o = JSON.parse(jsonString);
+
+    // Handle non-exception-throwing cases:
+    // Neither JSON.parse(false) or JSON.parse(1234) throw errors, hence the type-checking,
+    // but... JSON.parse(null) returns null, and typeof null === "object",
+    // so we must check for that, too. Thankfully, null is falsey, so this suffices:
+    if (o && typeof o === "object") {
+      return o;
+    }
+  } catch (e) {
+    return false;
+  }
+
+  return false;
+}
+
+@customElement("mqtt-device-debug-info-card")
+class MQTTDeviceDebugInfoCard extends LitElement {
+  @property() public hass!: HomeAssistant;
+  @property() public debugInfo?: MQTTDeviceDebugInfo;
+
+  protected render(): TemplateResult {
+    return html`
+      <ha-card header="Lab">
+        <div class="card-content">
+          ${this.debugInfo && this.debugInfo.entities.length > 0
+            ? html`
+                Entities
+                <ul>
+                  ${this.debugInfo.entities.map(
+                    (entity) => html`
+                      <li>
+                        '${computeStateName(
+                          this.hass.states[entity.entity_id]
+                        )}'
+                        (<code>${entity.entity_id}</code>)
+                        <br />Discovery topic:
+                        <code>${entity.discovery_data.discovery_topic}</code>
+                        <details>
+                          <summary>
+                            Discovery payload (JSON)
+                          </summary>
+                          <pre>
+${JSON.stringify(entity.discovery_data.discovery_payload, null, 2)}</pre
+                          >
+                        </details>
+                        <details>
+                          <summary>
+                            Discovery payload (YAML)
+                          </summary>
+                          <pre>
+${safeDump(entity.discovery_data.discovery_payload)}</pre
+                          >
+                        </details>
+                        Subscribed topics
+                        <ul>
+                          ${entity.topics.map(
+                            (topic) => html`
+                              <li>
+                                <code>${topic.topic}</code>
+                                <details>
+                                  <summary>
+                                    ${topic.messages.length} most recent
+                                    received message(s)
+                                  </summary>
+                                  <ul>
+                                    ${topic.messages.map(
+                                      (message) => html`
+                                        <li>
+                                          ${tryParseJSON(message)
+                                            ? html`
+                                                <pre>
+${safeDump(JSON.parse(message))}</pre
```

Comment:
This is very inefficient. You are now parsing it as JSON twice. 

I also think that this page will be very expensive to load because of all the parsing/rendering. This is fine, but we should remove `hass` as a `@property` or else we will keep re-rendering this dialog on every state change.

Reply:
> This is very inefficient. You are now parsing it as JSON twice.
Yeah, it's silly, will try to rewrite it.

> we should remove hass as a @property

Does it mean:
```diff
 class DialogMQTTDeviceDebugInfo extends LitElement {
-   @property() public hass!: HomeAssistant;
+  public hass!: HomeAssistant;
```
?

Reply:
Yes, that will prevent an update of `hass` to trigger a re-render.

Reply:
Thanks, improved now.
It could be further improved by not parsing and rendering unless the `<details>` is expanded.